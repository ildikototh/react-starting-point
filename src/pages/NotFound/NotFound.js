import React from 'react';
import Layout from '../../layouts/Layout';

const NotFound = () => (
  <Layout>
    <div>
      <h2>Page not found</h2>
      <p>Sorry, the page you&apos;re looking for can&apos;t be found</p>
    </div>
  </Layout>
);

export default NotFound;
