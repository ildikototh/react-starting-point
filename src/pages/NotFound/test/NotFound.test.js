import React from 'react';
import { shallow } from 'enzyme';
import NotFound from '../NotFound';

describe('<NotFound />', () => {
  describe('structure', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<NotFound />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });


    it('should render a connected <Layout />', () => {
      expect(wrapper).toContainMatchingElement('Layout');
    });


    it('should contain a heading and paragraph', () => {
      expect(wrapper).toContainMatchingElement('h2');
      expect(wrapper).toContainMatchingElement('p');
    });
  });
});
