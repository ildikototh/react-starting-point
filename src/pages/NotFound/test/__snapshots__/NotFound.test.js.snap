// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> structure should match the snapshot 1`] = `
<Layout
  className=""
>
  <div>
    <h2>
      Page not found
    </h2>
    <p>
      Sorry, the page you're looking for can't be found
    </p>
  </div>
</Layout>
`;
